packages/tools/src/tools/annotation/SplineROITool.ts

```diff
@@ -316,11 +323,21 @@ class SplineROITool extends ContourSegmentationBaseTool {
 
     if (newAnnotation) {
       const eventType = Events.ANNOTATION_COMPLETED;
+      const { viewportId, renderingEngineId } = enabledElement;
       const eventDetail: AnnotationCompletedEventDetail = {
         annotation,
+        viewportId,
+        renderingEngineId,
+        changeType: ChangeTypes.Completed,
       };
 
       triggerEvent(eventTarget, eventType, eventDetail);
+    } else {
+      this.triggerAnnotationModified(
+        annotation,
+        enabledElement,
+        ChangeTypes.HandlesUpdated
```

Comment:
The handles updated change is required to prevent updating the interpolated slices too often - it will get fired when the mouse is released rather than on every little move.